app/models/amusement_park.rb

```diff
@@ -1,3 +1,12 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+  has_many :mechanics, through: :rides
+
+  def find_mechanics
+    AmusementPark.joins(:mechanics).where("amusement_parks.id = ?", self.id).distinct.pluck("mechanics.name")
+  end
+
+  def ride_avg_experience
+    AmusementPark.joins(:mechanics).select("avg(mechanics.years_experience) AS avg_experience, rides.name AS ride_name").where("amusement_parks.id = ?", self.id).group(:ride_name).order(:avg_experience)
+  end
 end
```

Comment:
I really like that you added the `mechanics` relationship on line 3. However, I don't see that you're using it! For example, you could probably refactor the first method, `find_mechanics` to something like: 

```ruby
mechanics.select("mechanics.name").distinct
```